tutorials/Working_with_fNIRS_data.py

```diff
@@ -0,0 +1,124 @@
+# concentrations using the modified Lambert-Beer model.
+# 
+# In the following example we have data from 20 channels plus one events
+# channel and we end up # with 20 HbO time series, 20 HbR time series and one
+# events time series.
+
+###############################################################################
+# Often fNIRS data contain time series for oxyhemoglobin (HbO) and
+# deoxyhemoglobin (HbR) concentrations and you want to process them both. Using
+# the *channel types* field in a mne.Info class instance, you can assign either
+#'hbo' or 'hbr' channel type to your channel.
+nChannels = 20 # number of physical channels
+sampling_frequency = 15.625
+channel_names_fnirs = ['HbO '+"%.2d" % i for i in range(1,nChannels+1)] +
+    ['HbR '+"%.2d" % i for i in range(1,nChannels+1)]
+channel_names = channel_names_fnirs + ['Events']
+channel_types = ['hbo' for i in range(nChannels)] +
+    ['hbr' for i in range(nChannels)] + ['stim']
+info = mne.create_info(ch_names=channel_names, sfreq=sampling_frequency,
+    ch_types=channel_types, montage=None)
+info['lowpass'] = 0.0
+info['highpass'] = 0.0
+
+###############################################################################
+# Import fNIRS data from numpy array using mne.io.RawArray.
+raw_data = mne.io.RawArray(data=raw_ndarray, info=info, first_samp=0,
+    verbose=None)
+
+###############################################################################
+# In our data array the last row contains event data, coded as different
+# in a time series synchronous with the fNIRS recording. Let's find such events
+# using mne.find_events and plot the events timeline with mne.viz.plot_events.
+events = mne.find_events(raw_data, stim_channel='Events', shortest_event=1)
+event_id = {'ISI':1, 'Condition1': 2, 'Condition2': 4, 'Condition3': 8,
+    'Pause':16}
+color = {1:'black', 2:'green', 4:'magenta', 8:'cyan', 16:'yellow'}
+mne.viz.plot_events(events, raw_data.info['sfreq'], raw_data.first_samp,
+    color=color, event_id=event_id)
+
+###############################################################################
+# Sometimes we want to filter out unwanted frequencies from time series. For
+# example, here we bandpass filter our data between 0.01 and 2 Hz.
+l_freq = 0.01 # high-pass filter cutoff ( __/¯¯¯ )
+h_freq = 2    #  low-pass filter cutoff ( ¯¯¯\__ )
+raw_data.filter(l_freq, h_freq)
+
+###############################################################################
+# In order to have a general glimpse of the dataset, I start by plotting all
+# channels into a single plot window. I choose to color-code HbO, HbR and
+# events data using red, blue and black respectively. I also mark color-coded
+# events positions, using the same colors previously used in
+# mne.viz.plot_events.
+scalings = dict(hbo=10e-6, hbr=10e-6, stim=1)
+fig_title = 'fNIRS Raw Bandpass filtered [' + str(l_freq) + ' Hz, '
+    + str(h_freq) + ' Hz]'
+plot_colors = dict(hbo='r', hbr='b', stim='k')
```

Comment:
Maybe we should just change the default colors in `raw.plot`. Are these standard colors for fNIRS?